test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -12480,6 +12480,33 @@ def forward(self, x):
 
         self.run_test(LerpModel(), torch.rand(5, 4, 3))
 
+    @common_utils.parametrize("input_dtype", [torch.cfloat, torch.float])
+    @skipIfUnsupportedMinOpsetVersion(9)
+    def test_print_tensor_within_torch_nn_module(self, input_dtype: torch.dtype):
+        class PrintTensorOnMyModel(torch.nn.Module):
+            def forward(self, x):
+                # 'print' has side effect calling 'resolve_conj' and 'resolve_neg'.
+                x_firsts = x[:, 0]
+                print(f"x_firsts: {x_firsts}")
+                # 'tolist' has side effect calling 'resolve_conj' and 'resolve_neg'.
+                # Annotation added to pass torch script.
+                _: List[float] = x.tolist()
+                return x_firsts
+
+        m = PrintTensorOnMyModel().eval()
+        x = torch.randn(10, 5, dtype=input_dtype)
+        if input_dtype == torch.cfloat:
+            with self.assertRaises(RuntimeError):
+                self.run_test(
+                    m,
+                    x,
+                )
```

Comment:
I would expect no errors now? Maybe have separate tests for `_conj` and `conj_physical`?

Reply:
As mentioned on a previous review, an exception is also raised for complex when shape inference is enabled (and it is enabled by default).

Reply:
No need to create a test that will execute the exact same code as the existing ones. They share implementation

Reply:
I wonder why it errors, is it because of a bug in shape inference?

Reply:
several jit passes do not expect complex numbers. for optimization-only passes, it would be safe to just return from the pass without making any change, preventing the failure. On the other hand, we are not interested in having ONNX graphs without shape inference (IIRC it is not part of the public API). We only use disable it for specific debugging scenarios